zio-http/src/main/scala/zio/http/Request.scala

```diff
@@ -92,18 +104,31 @@ object Request {
       Patch(self.addHeaders ++ that.addHeaders, self.addQueryParams ++ that.addQueryParams)
   }
 
-  def default(method: Method, url: URL, body: Body = Body.empty) =
-    Request(body, Headers.empty, method, url, Version.`HTTP/1.1`, Option.empty)
+  def delete(path: String): Request = Request(method = Method.DELETE, url = URL(Path(path)))
+
+  def delete(url: URL): Request = Request(method = Method.DELETE, url = url)
+
+  def get(path: String): Request = Request(method = Method.GET, url = URL(Path(path)))
+
+  def get(url: URL): Request = Request(method = Method.GET, url = url)
+
+  def head(path: String): Request = Request(method = Method.HEAD, url = URL(Path(path)))
+
+  def head(url: URL): Request = Request(method = Method.HEAD, url = url)
+
+  def options(path: String): Request = Request(method = Method.OPTIONS, url = URL(Path(path)))
+
+  def options(url: URL): Request = Request(method = Method.OPTIONS, url = url)
 
-  def delete(url: URL): Request = default(Method.DELETE, url)
+  def patch(path: String, body: Body): Request = Request(method = Method.PATCH, url = URL(Path(path)), body = body)
 
-  def get(url: URL): Request = default(Method.GET, url)
+  def patch(url: URL, body: Body): Request = Request(method = Method.PATCH, url = url, body = body)
 
-  def options(url: URL): Request = default(Method.OPTIONS, url)
+  def post(path: String, body: Body): Request = Request(method = Method.POST, url = URL(Path(path)), body = body)
```

Comment:
I think that there's an issue here. See https://github.com/zio/zio-http/issues/2308